cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishStartService.java

```diff
@@ -0,0 +1,188 @@
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package se.simonsoft.cms.publish.rest;
+
+import java.util.*;
+
+import javax.inject.Inject;
+
+import com.fasterxml.jackson.databind.ObjectReader;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import se.simonsoft.cms.item.CmsItem;
+import se.simonsoft.cms.item.CmsItemId;
+import se.simonsoft.cms.item.command.CommandRuntimeException;
+import se.simonsoft.cms.publish.config.PublishConfiguration;
+import se.simonsoft.cms.publish.config.PublishExecutor;
+import se.simonsoft.cms.publish.config.command.PublishWebhookCommandHandler;
+import se.simonsoft.cms.publish.config.databinds.config.PublishConfig;
+import se.simonsoft.cms.publish.config.databinds.job.*;
+import se.simonsoft.cms.publish.config.databinds.profiling.PublishProfilingRecipe;
+import se.simonsoft.cms.publish.config.databinds.profiling.PublishProfilingSet;
+import se.simonsoft.cms.publish.config.item.CmsItemPublish;
+import se.simonsoft.cms.release.ProfilingSet;
+import se.simonsoft.cms.release.ReleaseProperties;
+import se.simonsoft.cms.release.translation.CmsItemTranslation;
+import se.simonsoft.cms.release.translation.TranslationLocalesMapping;
+import se.simonsoft.cms.release.translation.TranslationTracking;
+import se.simonsoft.cms.reporting.CmsItemLookupReporting;
+
+public class PublishStartService {
+
+	private final PublishWebhookCommandHandler publishWebhookCommandHandler;
+	private final CmsItemLookupReporting lookupReporting;
+	private final TranslationTracking translationTracking;
+	private final PublishConfiguration publishConfiguration;
+	private final PublishExecutor publishExecutor;
+	private final PublishJobFactory jobFactory;
+	ObjectReader profilingSetReader;
+
+	private static final Logger logger = LoggerFactory.getLogger(PublishStartService.class);
+
+	@Inject
+	public PublishStartService(
+			PublishWebhookCommandHandler publishWebhookCommandHandler,
+			CmsItemLookupReporting lookupReporting,
+			TranslationTracking translationTracking,
+			PublishConfiguration publishConfiguration,
+			PublishExecutor publishExecutor,
+			PublishJobFactory jobFactory,
+			ObjectReader reader) {
+
+		this.publishWebhookCommandHandler = publishWebhookCommandHandler;
+		this.lookupReporting = lookupReporting;
+		this.translationTracking = translationTracking;
+		this.publishConfiguration = publishConfiguration;
+		this.publishExecutor = publishExecutor;
+		this.jobFactory = jobFactory;
+		this.profilingSetReader = reader.forType(PublishProfilingSet.class);
+	}
+
+
+	public LinkedHashMap<String, String> doPublishStartItem(CmsItemId itemId, PublishStartOptions options) {
+
```

Comment:
Add validation of startpathname and startcustom here.

Probably need to inject ObjectWriter above.

Reply:
Two things:
 - The placeholder is incorrect for String.format. I created a test in scenariontesting.
 - I realized that we should have a similar validation for startprofiling, go for 100 kB as well. So user can spend their 110kB on startcustom, startprofiling or a mix thereof.

Reply:
Damn, you're right. Done.